app/src/main/java/com/monkeyteam/chimpagne/ui/components/EventCard.kt

```diff
@@ -0,0 +1,116 @@
+import com.monkeyteam.chimpagne.ui.theme.ChimpagneTypography
+
+@Composable
+fun EventCard(event: ChimpagneEvent, modifier: Modifier = Modifier, onClick: () -> Unit) {
+  Card(
+      modifier =
+          modifier
+              .padding(horizontal = 12.dp, vertical = 8.dp)
+              .clip(RoundedCornerShape(12.dp))
+              .border(
+                  width = 3.dp,
+                  color = MaterialTheme.colorScheme.primary,
+                  shape = RoundedCornerShape(12.dp))
+              .shadow(4.dp)
+              .fillMaxWidth()
+              .clickable { onClick() }
+              .aspectRatio(1.9f),
+      shape = RoundedCornerShape(12.dp),
+      colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)) {
+        Column(modifier = Modifier.fillMaxSize()) {
+          Box(modifier = Modifier.fillMaxWidth().weight(1f)) {
+            ImageWithBlackFilterOverlay(event.image, false)
+            // Adding the status overlay on top of the image
+            Row(
+                horizontalArrangement = Arrangement.End,
+                modifier =
+                    Modifier.align(Alignment.TopEnd)
+                        .padding(top = 12.dp, end = 12.dp)
+                        .background(
+                            color =
+                                Color.Black.copy(alpha = 0.8f), // Semi-transparent black background
+                            shape = RoundedCornerShape(100) // Rounded corners
+                            )
+                        .padding(
+                            start = 8.dp,
+                            end = 8.dp,
+                            top = 4.dp,
+                            bottom = 4.dp) // Padding inside the background
+                ) {
+                  Text(
+                      text =
+                          if (event.public) stringResource(id = R.string.public_string)
+                          else stringResource(id = R.string.private_string),
+                      style = MaterialTheme.typography.bodyLarge,
+                      color = Color.White, // Ensure the text is visible against the background
+                      modifier = Modifier.padding(end = 8.dp))
+                  Icon(
+                      imageVector =
+                          if (event.public) Icons.Rounded.Public else Icons.Rounded.PublicOff,
+                      contentDescription = if (event.public) "Public Event" else "Private Event",
+                      tint = Color.White // Ensure the icon is visible against the background
+                      )
+                }
+          }
+          Column(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .padding(top = 8.dp, start = 16.dp, end = 16.dp, bottom = 16.dp)) {
+                Text(
+                    text = event.title,
+                    style = ChimpagneTypography.titleLarge,
```

Comment:
We discussed it today, and you already moved to ChimpagneTypography. Fast adaptation is the key to success 😎